common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java

```diff
@@ -113,6 +118,13 @@ public BrokerRequest brokerRequestFromAcquireTokenParameters(@NonNull final Inte
                                 AuthorizationAgent.WEBVIEW.name()
                 ).authenticationScheme(parameters.getAuthenticationScheme())
                 .powerOptCheckEnabled(parameters.isPowerOptCheckEnabled())
+                .spanContext(SerializableSpanContext.builder()
+                        .traceId(SpanExtension.current().getSpanContext().getTraceId())
+                        .spanId(SpanExtension.current().getSpanContext().getSpanId())
+                        .traceFlags(SpanExtension.current().getSpanContext().getTraceFlags().asByte())
+                        .parentSpanName(OTelUtility.getCurrentSpanName())
+                        .build()
+                )
```

Comment:
so shahzaib, do we have to add this span context change to every type of request made to broker? Asking because Pallav is adding a new method to create a broker request for DCF.

Reply:
@somalaya span context is an optional field so it is not required to be present. If we want to capture telemetry for DCF then Pallav should add it in that method.